### PR TITLE
MDEV-22629: Remove fts_indexes field from struct fts_update_t

### PR DESCRIPTION
**Pull request by t1meshift**
I noticed that `fts_indexes` field in `fts_update_t` was not used at all.

### CODE REVIEW

**Comment by an3l** [comment]
Hi @t1meshift ,
Thanks for your contribution.

Similar to other open source projects, the MariaDB Foundation needs to have shared ownership of all code that is included in the MariaDB distribution. The easiest way to achieve this is by submitting your code under the BSD-new license.

The other alternative is to sign the code contribution agreement which can be found here: https://mariadb.com/kb/en/mariadb/mca/

Please indicate in a comment below, or update your PR comment, that you are contributing your new code of the whole pull request, including one or several files that are either new files or modified ones, under the BSD-new license or that you have filled out the contribution agreement and sent it.

Thanks,
Anel

**Comment by t1meshift** [comment]
Done.

**Comment by kevgs** [comment]
At the beginning of every file you can see this https://github.com/mariadb/server/blob/10.2/storage/innobase/fts/fts0opt.cc#L4
Here `2016, 2019,` stands for time interval. You're updating this file in 2020, so, please, fix the last year in the files you modify.

**Comment by t1meshift** [comment]
Fixed it.

**Comment by dr-m** [comment]
> At the beginning of every file you can see this https://github.com/mariadb/server/blob/10.2/storage/innobase/fts/fts0opt.cc#L4
> Here `2016, 2019,` stands for time interval. You're updating this file in 2020, so, please, fix the last year in the files you modify.

That advice applies to the employees of MariaDB Corporation. I do not know if we should request external contributors to update MariaDB copyright years. I do not forbid it, though. @t1meshift, it would be better to squash and force-push, based on the latest 10.2 branch.

I have one more suggestion: remove `fts_update_t` altogether and replace it with `doc_id_t`. While doing that, consider replacing `fts_update_t* array` with `doc_id_t* updates`.

File: storage/innobase/include/fts0types.h [hunk 1]
```diff
@@ -321,8 +307,7 @@ fts_ranking_doc_id_cmp(
 /******************************************************************//**
 Compare two fts_update_t instances doc_ids. */
 UNIV_INLINE
-int
-fts_update_doc_id_cmp(
+int doc_id_cmp(
```
Comment (kevgs) [inline storage/innobase/include/fts0types.h:310]: I don't like it too, but let's keep `fts_` prefix here for consistency.

File: storage/innobase/include/fts0types.ic [hunk 2]
```diff
@@ -79,19 +79,18 @@ fts_ranking_doc_id_cmp(
...
-int
-fts_update_doc_id_cmp(
+int doc_id_cmp(
 /*==================*/
 	const void*	p1,			/*!< in: id1 */
 	const void*	p2)			/*!< in: id2 */
 {
-	const fts_update_t*	up1 = (const fts_update_t*) p1;
-	const fts_update_t*	up2 = (const fts_update_t*) p2;
+	const doc_id_t*	up1 = (const doc_id_t*) p1;
```
Comment (kevgs) [inline storage/innobase/include/fts0types.ic:90]: Please, use `static_cast<>` here.

File: storage/innobase/include/fts0types.ic [hunk 3]
```diff
@@ -79,19 +79,18 @@ fts_ranking_doc_id_cmp(
...
+	const doc_id_t*	up2 = (const doc_id_t*) p2;
 
-	return((int)(up1->doc_id - up2->doc_id));
+	return((int)(up1 - up2));
```
Comment (kevgs) [inline storage/innobase/include/fts0types.ic:93]: And here `static_cast<>` too.

Comment (t1meshift) [reply storage/innobase/include/fts0types.ic]: There was a bigger problem, actually :)

Comment (kevgs) [reply storage/innobase/include/fts0types.ic]: Good that it was caught! But apparently, it wasn't covered by tests.

File: storage/innobase/fts/fts0opt.cc [hunk 4]
```diff
@@ -1022,7 +1021,7 @@ Do a binary search for a doc id in the array
 int
 fts_bsearch(
 /*========*/
-	fts_update_t*	array,	/*!< in: array to sort */
+	doc_id_t*	updates,	/*!< in: array to sort */
```
Comment (Thirunarayanan) [inline storage/innobase/fts/fts0opt.cc:1024]: you can avoid the renaming of the variable here.

File: storage/innobase/fts/fts0opt.cc [hunk 5]
```diff
@@ -1074,9 +1073,9 @@ fts_optimize_lookup(
 {
 	int		pos;
 	int		upper = static_cast<int>(ib_vector_size(doc_ids));
-	fts_update_t*	array = (fts_update_t*) doc_ids->data;
+	doc_id_t*	updates = (doc_id_t*) doc_ids->data;
```
Comment (Thirunarayanan) [inline storage/innobase/fts/fts0opt.cc:1076]: same goes here

**Comment by t1meshift** [comment]
@Thirunarayanan you're right. I thought `updates` naming was OK since there was `fts_update_t*` type.

**Comment by Thirunarayanan** [comment]
I have created the MDEV for this issue (https://jira.mariadb.org/browse/MDEV-22629).
Please squash all the commit into single one and write commit message.
In commit message, title should be  Also MDEV# and bug summary. Once you have done that, I can test it in builbot.

**Comment by t1meshift** [comment]
@Thirunarayanan done.

**Pull request merged**